src/app/channels/channels-routing.module.ts

```diff
@@ -0,0 +1,39 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+import { ChannelComponent } from './channel/channel.component';
+import { ChannelsComponent } from './channels.component';
+import { ChannelAddComponent } from './channel-add/channel-add.component';
+import { ChannelSearchComponent } from './channel-search/channel-search.component';
+
+const routes: Routes = [
+  {
+    path: 'all',
+    component: ChannelsComponent,
+  },
+  {
+    path: 'user',
+    component: ChannelsComponent,
+  },
+  {
+    path: 'upload',
+    component: ChannelsComponent,
+  },
+  {
+    path: 'create',
+    component: ChannelAddComponent,
+  },
+  {
+    path: 'search',
+    component: ChannelSearchComponent,
+  },
+  {
+    path: 'channel/:id',
```

Comment:
I think its better to use RESTAPI like routes here...
Instead of `/channels/channel/:id` can use only `/channels/:id`
Instead of `/channels/all` can use only `/channels`

Reply:
Done